# GenosÂ GenomicÂ FoundationÂ Model

<p align="center">
  English | <a href="README.zh.md">ä¸­æ–‡</a>
</p>

## 1.Â ModelÂ Name

Genos-1.2B-32KÂ /Â Genos-10B

![image.png](https://alidocs.oss-cn-zhangjiakou.aliyuncs.com/res/AJdl65AJbjamvOke/img/a5308220-330d-4e76-84e5-d773856897e6.png)

![image](https://alidocs.oss-cn-zhangjiakou.aliyuncs.com/res/meonarb8PVbQ0qXx/img/ff899db6-5f29-4e54-b3a9-8cf1d0e4ef0f.png)

## 2.BriefÂ Description

HumanÂ GenomeÂ IntelligenceÂ "System":Â GenosÂ VeryÂ LargeÂ ScaleÂ Pre-TrainingÂ HumanÂ GenomeÂ BaseÂ Model

**\[ModelÂ architectureÂ andÂ technologyÂ breakthrough\]**Â asÂ theÂ leadingÂ basicÂ modelÂ inÂ theÂ fieldÂ ofÂ humanÂ genome,Â GenosÂ reliesÂ onÂ massiveÂ high-qualityÂ genomicÂ benchmarkÂ dataÂ forÂ in-depthÂ training,ItsÂ innovationÂ liesÂ inÂ theÂ breakthroughÂ realizationÂ ofÂ theÂ humanÂ genomeÂ sequenceÂ upÂ toÂ millionÂ baseÂ pairsÂ ofÂ ContextÂ modelingÂ ability.Â ThroughÂ single-baseÂ resolutionÂ learning,Â theÂ modelÂ successfullyÂ analyzesÂ theÂ deepÂ sequenceÂ rulesÂ andÂ functionalÂ characteristicsÂ hiddenÂ inÂ theÂ genome,Â andÂ constructsÂ anÂ intelligentÂ bridgeÂ connectingÂ geneticÂ informationÂ andÂ lifeÂ activities.Â TheÂ currentÂ versionÂ includesÂ twoÂ configurationsÂ ofÂ 1.2Â billionÂ parametersÂ andÂ 10Â billionÂ parameters,TheÂ advancedÂ hybridÂ expertÂ (MoE)Â architectureÂ isÂ adoptedÂ toÂ realizeÂ theÂ optimalÂ allocationÂ ofÂ computingÂ resourcesÂ throughÂ dynamicÂ routingÂ mechanism,Â whichÂ significantlyÂ improvesÂ theÂ performanceÂ ofÂ theÂ modelÂ inÂ complexÂ regulatoryÂ networkÂ analysis.

**\[FunctionÂ moduleÂ andÂ scientificÂ value\]**Â asÂ theÂ "linguist"Â ofÂ genome,Â GenosÂ hasÂ theÂ coreÂ abilityÂ toÂ accuratelyÂ identifyÂ keyÂ functionalÂ elements,ItÂ canÂ deeplyÂ analyzeÂ theÂ cascadeÂ effectÂ ofÂ microÂ geneÂ variationÂ onÂ transcriptionalÂ regulatoryÂ network.Â ItsÂ innovationÂ isÂ reflectedÂ inÂ theÂ predictionÂ accuracyÂ ofÂ regulatoryÂ elementsÂ inÂ theÂ non-codingÂ region,Â whichÂ breaksÂ throughÂ theÂ limitationsÂ ofÂ traditionalÂ methods,Â canÂ dynamicallyÂ simulateÂ theÂ potentialÂ impactÂ ofÂ mutationÂ sitesÂ onÂ RNAÂ expressionÂ profiles,Â andÂ traceÂ theÂ molecularÂ pathÂ ofÂ phenotypeÂ formation.Â OnÂ thisÂ basis,Â theÂ researchÂ teamÂ developedÂ aÂ modularÂ applicationÂ interface,BuildÂ aÂ full-chainÂ researchÂ systemÂ ofÂ "prediction-interpretation-verification.Â ByÂ introducingÂ anÂ interpretableÂ enhancementÂ mechanism,Â theÂ modelÂ notÂ onlyÂ providesÂ highÂ confidenceÂ predictionÂ results,Â butÂ alsoÂ revealsÂ theÂ keyÂ nodesÂ andÂ pathwaysÂ inÂ theÂ regulatoryÂ network,Â providingÂ aÂ newÂ researchÂ paradigmÂ forÂ molecularÂ mechanismÂ analysis.

**\[OpenÂ EcologyÂ andÂ ClinicalÂ Transformation\]**Â AdheringÂ toÂ theÂ conceptÂ ofÂ openÂ science,GenosÂ hasÂ deployedÂ cloudÂ reasoningÂ servicesÂ onÂ theÂ HuadaÂ DCSCloudÂ platformÂ toÂ buildÂ aÂ "cloudÂ lab"Â forÂ genomicÂ intelligenceÂ analysisÂ ".Â ResearchersÂ canÂ uploadÂ dataÂ throughÂ anÂ intuitiveÂ interfaceÂ toÂ obtainÂ aÂ full-processÂ analysisÂ fromÂ mutationÂ functionÂ annotationÂ toÂ phenotypeÂ prediction,Â completelyÂ freeÂ fromÂ theÂ constraintsÂ ofÂ localÂ computingÂ powerÂ andÂ algorithmÂ deployment.Â ThisÂ decentralizedÂ computingÂ powerÂ supportÂ model,ItÂ enablesÂ researchersÂ worldwideÂ toÂ shareÂ theÂ predictiveÂ powerÂ ofÂ leadingÂ modelsÂ andÂ accelerateÂ theÂ transitionÂ fromÂ genomicÂ discoveryÂ toÂ clinicalÂ applications.Â AsÂ theÂ modelÂ continuesÂ toÂ optimizeÂ andÂ iterate,Â itsÂ applicationÂ potentialÂ inÂ precisionÂ medicine,Â groupÂ healthÂ monitoringÂ andÂ developmentalÂ biologyÂ willÂ continueÂ toÂ beÂ released,Â layingÂ aÂ keyÂ foundationÂ forÂ buildingÂ anÂ activeÂ healthÂ managementÂ systemÂ coveringÂ theÂ wholeÂ lifeÂ cycle.

**\[PhilosophyÂ ofÂ ScienceÂ andÂ FutureÂ Prospects\]**Â TheÂ birthÂ ofÂ GenosÂ marksÂ aÂ majorÂ innovationÂ inÂ theÂ researchÂ paradigmÂ ofÂ genomics.Â ItÂ isÂ notÂ onlyÂ aÂ breakthroughÂ toolÂ forÂ computationalÂ biology,Â butÂ alsoÂ anÂ intelligentÂ carrierÂ forÂ decodingÂ theÂ mysteriesÂ ofÂ life.Â ByÂ fusingÂ multi-omicsÂ dataÂ withÂ phenotypicÂ information,Â theÂ modelÂ isÂ reshapingÂ ourÂ cognitiveÂ frameworkÂ forÂ understandingÂ gene-environmentÂ interactions.InÂ theÂ future,Â withÂ theÂ improvementÂ ofÂ cross-modalÂ learningÂ ability,Â GenosÂ isÂ expectedÂ toÂ becomeÂ aÂ "translator"Â connectingÂ geneticÂ codeÂ andÂ lifeÂ phenomena,Â openingÂ upÂ aÂ newÂ researchÂ dimensionÂ inÂ theÂ fieldsÂ ofÂ diseaseÂ earlyÂ warning,Â drugÂ targetÂ discoveryÂ andÂ syntheticÂ biology,Â andÂ finallyÂ realizingÂ theÂ paradigmÂ leapÂ fromÂ "genomics"Â toÂ "functionalÂ omics.Â ThisÂ landmarkÂ technologicalÂ breakthrough,ItÂ isÂ pushingÂ mankindÂ steadilyÂ towardsÂ aÂ newÂ eraÂ ofÂ precisionÂ medicine.

## 3.Â PerformanceÂ comparisonÂ chart

![image.png](https://alidocs.oss-cn-zhangjiakou.aliyuncs.com/res/eYVOL5jwEmY8Glpz/img/6b844694-626c-41a5-bc02-4c4ba0e64970.png)

## 4.Â ModelÂ Description

Genos:Â ParadigmÂ InnovationÂ ofÂ GenomicÂ IntelligenceÂ AnalysisÂ Engine\]

**DataÂ Base:Â constructÂ globalÂ genomeÂ diversityÂ map**Â thisÂ studyÂ integratesÂ theÂ standardizedÂ dataÂ ofÂ topÂ internationalÂ genomicsÂ cohortsÂ suchÂ asÂ humanÂ pan-GenomeÂ ReferenceÂ ConsortiumÂ (HPRC),Â HumanÂ genomeÂ structuralÂ variationÂ mappingÂ projectÂ (HGSVC),Â etc,AÂ multi-sourceÂ HeterogeneousÂ genomeÂ dataÂ setÂ coveringÂ fiveÂ continentsÂ ofÂ Europe,Â Asia,Â Africa,Â theÂ UnitedÂ StatesÂ andÂ AustraliaÂ andÂ containingÂ hundredsÂ ofÂ casesÂ ofÂ genome-wideÂ telomereÂ toÂ TelomereÂ (telomere-to-telomere)Â assemblyÂ wasÂ constructed.Â ByÂ implementingÂ aÂ strictÂ qualityÂ controlÂ process-includingÂ baseÂ qualityÂ segmentationÂ filtering,Â structuralÂ variationÂ standardizedÂ mappingÂ andÂ populationÂ geneticÂ backgroundÂ correction-toÂ ensureÂ thatÂ theÂ dataÂ setÂ achievesÂ clinical-levelÂ accuracyÂ atÂ singleÂ nucleotideÂ resolutionÂ (singleÂ nucleotideÂ resolution),LayÂ aÂ solidÂ foundationÂ forÂ cross-ethnicÂ generalization.

**ArchitectureÂ Innovation:Â GenomeÂ DecodingÂ RevolutionÂ ofÂ HybridÂ ExpertÂ Network**Â BasedÂ onÂ theÂ deepÂ evolutionÂ ofÂ TransformerÂ architecture,Â thisÂ systemÂ innovativelyÂ constructsÂ HierarchicalÂ HybridÂ expertÂ networkÂ (HMoE),ItsÂ coreÂ breakthroughÂ isÂ reflectedÂ in:

1.  **SingleÂ nucleotideÂ resolutionÂ modelingÂ ofÂ ultra-longÂ sequences**Â SuccessfullyÂ overcomeÂ theÂ modelingÂ challengeÂ ofÂ million-baseÂ sequencesÂ byÂ introducingÂ ultra-longÂ sequenceÂ parameterizationÂ strategies,Â multi-dimensionalÂ tensorÂ parallelÂ computingÂ andÂ multi-scaleÂ attentionÂ mechanisms.Â TheÂ innovativeÂ fractalÂ attentionÂ moduleÂ effectivelyÂ alleviatesÂ theÂ local-globalÂ dependencyÂ decayÂ problemÂ inÂ veryÂ longÂ contextsÂ (10Â ^Â 6Â bp),TheÂ cooperativeÂ analysisÂ ofÂ singleÂ baseÂ variationÂ (SNV)Â andÂ structuralÂ variationÂ (SV)Â wasÂ realized.
    
2.  **TrainingÂ StabilityÂ OptimizationÂ System**Â AnÂ expertÂ loadÂ balancingÂ mechanismÂ hasÂ beenÂ developedÂ forÂ theÂ distributionÂ ofÂ lowÂ entropyÂ characteristicsÂ uniqueÂ toÂ genomicÂ data.Â ThroughÂ theÂ collaborativeÂ optimizationÂ ofÂ gradientÂ clippingÂ andÂ expertÂ selectionÂ strategy,Â theÂ problemÂ ofÂ expertÂ moduleÂ loadÂ imbalanceÂ causedÂ byÂ smallÂ vocabularyÂ sizeÂ (4Â bases)Â isÂ successfullyÂ solved.
    
3.  **DynamicÂ ExpertÂ ActivationÂ Architecture**Â releasedÂ twoÂ versionsÂ ofÂ theÂ intelligentÂ engine:Â TheÂ EliteÂ versionÂ (1.2Â billionÂ parameters)Â andÂ theÂ flagshipÂ versionÂ (10Â billionÂ parameters),Â bothÂ ofÂ whichÂ supportÂ Million-levelÂ ultra-longÂ sequenceÂ inference.Â ItsÂ originalÂ DynamicÂ RoutingÂ AlgorithmÂ (DynamicÂ RoutingÂ Algorithm)Â canÂ beÂ basedÂ onÂ theÂ epigeneticÂ characteristicsÂ ofÂ theÂ inputÂ sequence,Â populationÂ geneticÂ backgroundÂ andÂ otherÂ meta-information,ActivateÂ theÂ relevantÂ expertÂ moduleÂ inÂ realÂ time.
    

TheÂ birthÂ ofÂ GenosÂ marksÂ theÂ qualitativeÂ changeÂ ofÂ genomicsÂ researchÂ paradigmÂ fromÂ "dataÂ accumulation"Â toÂ "intelligentÂ analysis",Â andÂ providesÂ aÂ revolutionaryÂ intelligentÂ decisionÂ supportÂ platformÂ forÂ theÂ designÂ ofÂ individualizedÂ diagnosisÂ andÂ treatmentÂ plan,Â diseaseÂ riskÂ predictionÂ andÂ preciseÂ interventionÂ strategyÂ formulation.Â ThisÂ breakthrough,ItÂ isÂ graduallyÂ pushingÂ humanÂ interpretationÂ ofÂ theÂ lifeÂ codeÂ toÂ aÂ newÂ dimensionÂ ofÂ single-baseÂ accuracy.

| **Parameters** | **1.2B** | **10B** |
| --- | --- | --- |
| **NumberÂ ofÂ Layers**Â <br>**(DenseÂ layerÂ included)** | 1.25B | 10.27B |
| **MoEÂ HiddenÂ DimensionÂ (perÂ Expert)** | 1024 | 4096 |
| **NumberÂ ofÂ AttentionÂ Heads** | 4096 | 8192 |
| **VocabularyÂ Size** | 128(padded) | 256(padded) |
| **TrainedÂ Tokens** | 1500Â B | 2000Â B |

## 5.Â DeploymentÂ andÂ Use

*   **CurrentÂ status**:Â RÂ &Â DÂ andÂ optimizationÂ phase,Â supportingÂ internalÂ scientificÂ research.
    
*   **HardwareÂ requirements**:Â SupportÂ mainstreamÂ GPUÂ environment,Â noÂ specialÂ hardwareÂ restrictions
    

**Usage**:Â providesÂ modelÂ weightsÂ forÂ downstreamÂ taskÂ tuning,SupportÂ sequenceÂ embeddingÂ generation,Â multi-groupÂ dataÂ generation,Â variationÂ effectÂ predictionÂ andÂ otherÂ tasks.

## 6.Â PerformanceÂ evaluation

GenosÂ geneÂ pedestalÂ modelÂ evaluationÂ system

TheÂ purposeÂ ofÂ thisÂ evaluationÂ systemÂ isÂ toÂ systematicallyÂ evaluateÂ theÂ comprehensiveÂ abilityÂ ofÂ GenosÂ ModelÂ inÂ genomeÂ sequenceÂ analysis,Â transcriptionÂ effectÂ predictionÂ andÂ biomedicalÂ downstreamÂ applications.Â OurÂ evaluationÂ focusesÂ notÂ onlyÂ onÂ theÂ model'sÂ scoreÂ onÂ theÂ standardÂ benchmarkÂ dataset,PayÂ moreÂ attentionÂ toÂ itsÂ potentialÂ toÂ solveÂ real-worldÂ biomedicalÂ problems.Â ItÂ isÂ dividedÂ intoÂ threeÂ types:Â **shortÂ sequenceÂ evaluation**,Â \*\*longÂ sequenceÂ evaluation,Â andÂ mutationÂ hotÂ spotÂ prediction.Â \*\*TheÂ threeÂ tasksÂ wereÂ toÂ assessÂ theÂ model'sÂ identificationÂ andÂ understandingÂ ofÂ geneÂ elements,Â itsÂ understandingÂ ofÂ long-rangeÂ regulation,Â andÂ itsÂ abilityÂ toÂ captureÂ theÂ susceptibilityÂ ofÂ localÂ variationÂ basedÂ onÂ sequenceÂ featuresÂ alone,FurtherÂ examineÂ whetherÂ theÂ modelÂ hasÂ theÂ abilityÂ toÂ depictÂ theÂ signalÂ relatedÂ toÂ populationÂ differentiationÂ andÂ evolutionaryÂ history.

|  | **task** | **1.2b-8k** | **10b-1M** | **GENERator-3b** | **HyenaDNA-1M** | **NT-2.5b-multi** | **Evo2-7b** | **Evo2-40b** |
| --- | --- | --- | --- | --- | --- | --- | --- | --- |
| ShortÂ sequenceÂ evaluationÂ   <br>(sequenceÂ lengthÂ 200-600bp) | demo\_coding\_vs\_intergenomic\_seqs | 0.9717 | **0.9907** | 0.9855 | 0.9127 | 0.9763 | 0.9824 | **0.9886** |
|  | human\_enhancers\_cohn | **0.8723** | **0.8806** | 0.8181 | 0.7799 | 0.7873 | 0.7733 | 0.7756 |
|  | human\_ocr\_ensembl | **0.7730** | **0.7785** | 0.7270 | 0.6916 | 0.7285 | 0.7505 | 0.7635 |
|  | splice\_sites\_all | 0.7704 | 0.8064 | 0.8071 | 0.7110 | 0.8603 | **0.8747** | **0.9138** |
|  | H3 | **0.9413** | **0.9394** | 0.9163 | 0.8722 | 0.9371 | 0.9140 | 0.9311 |
|  | H3K36me3 | 0.7949 | 0.8300 | 0.8247 | 0.6787 | 0.8288 | **0.8615** | **0.8823** |
| MutationÂ hotÂ spotÂ evaluationÂ   <br>(sequenceÂ length:Â 8KÂ ~Â 128K) | CPC\_131072 | 0.9600 | **0.9886** | 0.9620 | **0.9735** |  |  |  |
|  | CPC\_32768 | 0.9331 | **0.9720** | 0.9237 | 0.9064 |  | 0.9504 | **0.9611** |
|  | CPC\_8192 | **0.9437** | **0.9547** | 0.9315 | 0.8914 |  | 0.9425 | 0.9401 |
| LongÂ sequenceÂ evaluationÂ   <br>(sequenceÂ length:Â 8K) | regulatory\_element\_enhancer\_8K | **0.7532** | **0.7536** | 0.7390 | 0.7282 |  | 0.7454 | 0.7527 |
|  | regulatory\_element\_promoter\_8K | 0.9252 | **0.9291** | 0.9195 | 0.8890 |  | **0.9255** | 0.9227 |
|  | variant\_effect\_causal\_eqtl\_8K | **0.7078** | 0.6973 | 0.6920 | 0.6887 |  | 0.7039 | **0.7054** |
|  | variant\_effect\_pathogenic\_clinvar\_8K | 0.6613 | **0.9298** | 0.7206 | 0.6117 |  | 0.7308 | **0.9167** |

## 7.Â CaseÂ Description

### CaseÂ 1:RNA-seqÂ DataÂ Generation

![image.png](https://alidocs.oss-cn-zhangjiakou.aliyuncs.com/res/meonarb8PVbQ0qXx/img/e12686fb-bc3f-4849-ac4c-6aa59f1fe859.png)

## TrainingÂ process

### 1.1Â PreprocessingÂ ofÂ DataÂ Preparation

1.  TheÂ trainingÂ dataÂ inÂ thisÂ caseÂ isÂ currentlyÂ derivedÂ fromÂ EncodeÂ andÂ GtexÂ (theÂ sameÂ asÂ AlphaGenome).Â ObtainÂ single-baseÂ transcriptomeÂ dataÂ ofÂ 667Â groupsÂ ofÂ samples,Â defineÂ theÂ dataÂ asÂ differentÂ batchesÂ accordingÂ toÂ differentÂ cellÂ typesÂ andÂ positiveÂ andÂ negativeÂ strands,Â andÂ giveÂ labels.TheÂ expressionÂ ofÂ multipleÂ samplesÂ inÂ eachÂ batchÂ isÂ averagedÂ toÂ obtainÂ single-baseÂ expressionÂ dataÂ ofÂ cellÂ typeÂ normalization,Â whichÂ isÂ maintainedÂ inÂ bigwigÂ format.
    
2.  BasedÂ onÂ theÂ modelÂ supportÂ length,Â referÂ toÂ theÂ RNA-seqÂ dataÂ ofÂ genomeÂ and.Â bw,Â interceptÂ accordingÂ toÂ theÂ 32kbÂ lengthÂ window,TheÂ twoÂ windowsÂ overlapÂ 16kbÂ toÂ ensureÂ theÂ repeatabilityÂ ofÂ theÂ trainingÂ data.
    
3.  Finally,Â theÂ pairsÂ ofÂ 32KBÂ genomeÂ andÂ transcriptomeÂ dataÂ areÂ providedÂ toÂ theÂ baseÂ modelÂ asÂ modelÂ inputÂ andÂ output,Â andÂ theÂ PearsonÂ correlationÂ coefficientÂ ofÂ theÂ predictionÂ vsÂ trueÂ expressionÂ isÂ usedÂ asÂ theÂ mainÂ lossÂ forÂ fine-tuningÂ training.
    

### 1.2Â NetworkÂ Architecture

```apl
+---------------------+
|   Input: input_ids  |
|     [Batch, Length] |
+---------------------+
           â†“
+-----------------------------+
|      Base Model (gLM)       |
|    â†’ last_hidden_state      |
|     [B, L, D]               |
+-----------------------------+
           â†“ (transpose)
+-----------------------------+
|   Transpose: [B, L, D] â†’    |
|       [B, D, L]             |
+-----------------------------+
           â†“
+--------------------------------------------------+
|                   CNN Head                       |
|  Sequential:                                     |
|  1. Conv1d(D â†’ D/4, k=3, p=1)                    |
|     â†’ BatchNorm â†’ GELU â†’ Dropout                 |
|                                                  |
|  2. Conv1d(D/4 â†’ D/16, k=3, p=2, dilation=2)     |
|     â†’ BatchNorm â†’ GELU â†’ Dropout                 |
|                                                  |
|  3. Conv1d(D/16 â†’ 1, k=1) â†’ squeeze               |
|                                                  |
|  Output: logits [B, L]                           |
+--------------------------------------------------+
           â†“
+-----------------------------------------+
|    Softplus Scaling:                    |
|    logits = F.softplus(logits) *        |
|             F.softplus(scale_param)     |
|    â†’ Ensures output > 0                 |
+-----------------------------------------+
           â†“
     â•­â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â•®
     â”‚ Training?   â”‚
     â•°â”€â”€â”€â”€â”€â”¬â”€â”€â”€â”€â”€â”€â”€â•¯
           â”‚ Yes
           â†“
+-----------------------------------------+
|   Loss Computation                      |
|   scaled_labels = target_scaling(       |
            labels, track_mean)           |
|                                         |
|   Supported:                            |
|   - MSE (selected)                      |
|   - Poisson                             |
|   - Tweedie                             |
|   - Poisson-Multinomial                 |
|                                         |
|   loss = L(logits, labels)              |
+-----------------------------------------+
           â”‚
     â”Œâ”€â”€â”€â”€â”€â”´â”€â”€â”€â”€â”€â”€â”
     â”‚ No / Eval  â”‚
     â•°â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â•¯
           â†“
+--------------------------------------+
|   Inference                          |
|   logits = predictions_scaling(      |   
|              logits, track_mean)     |
+--------------------------------------+
           â†“
+-----------------------------+
|         Output Dict         |
| { "loss": loss,             |
|   "logits": logits }        |
+-----------------------------+
```

## TrainingÂ parameters

*   CompleteÂ theÂ trainingÂ ofÂ 1~22Â chromosomesÂ andÂ fourÂ trajectoriesÂ modelingÂ separately.
    
*   1.2Â b-128kÂ modelÂ fullÂ amountÂ fineÂ adjustment
    
*   8Â nodesÂ 64Â cardsÂ (H100)Â trainingÂ 40Â rounds
    

```python
num_train_epochs=1,
per_device_train_batch_size=1,
per_device_eval_batch_size=1,
gradient_accumulation_steps=32,

dataloader_num_workers = 4,

learning_rate=5e-5,
lr_scheduler_type="cosine",
warmup_ratio=0.05,
weight_decay=0.01,
max_grad_norm=1.0,
optim="adafactor",

eval_strategy="epoch",
save_strategy="epoch",
eval_accumulation_steps=10,
save_total_limit=5,
save_safetensors=True,

fp16=False,
bf16=True,
half_precision_backend="auto",

logging_steps=1,
report_to="wandb",

ddp_find_unused_parameters=True,
seed=42
```
```markdown
09/23/2025 01:29:48 - INFO - root - âœ… [Distributed] åˆå§‹åŒ–æˆåŠŸï¼ rank=0, world_size=8, local_rank=0
09/23/2025 01:29:48 - INFO - root - âœ… [Distributed] åˆå§‹åŒ–æˆåŠŸï¼ rank=1, world_size=8, local_rank=1
09/23/2025 01:29:48 - INFO - root - âœ… [Distributed] åˆå§‹åŒ–æˆåŠŸï¼ rank=2, world_size=8, local_rank=2
09/23/2025 01:29:48 - INFO - root - âœ… Logging system initialized. Log file: output3__dlc/log/training_20250923_012945.log
09/23/2025 01:29:48 - INFO - root - âœ… [Distributed] åˆå§‹åŒ–æˆåŠŸï¼ rank=3, world_size=8, local_rank=3
09/23/2025 01:29:51 - INFO - root - ğŸŒ wandb: Logged in as: beneldor-zhejiang-lab
09/23/2025 01:29:51 - INFO - root - ğŸ“Š Project: RNA-Seq_Coverage_Track | Run Name: bs256_chr19_track27_epoch20_one-hot-mix-1.2b-128k_flash_attn-pai_mse_0923_dlc
09/23/2025 01:29:51 - INFO - root - ğŸš€ Run URL: None
09/23/2025 01:29:51 - INFO - root - ğŸ’¾ Local Dir: /your/path/RNASeqCoverageTrackPrediction_single-track/wandb/offline-run-20250923_012951-w0sac55n/files
09/23/2025 01:29:51 - INFO - root - ğŸš€ åŠ è½½é¢„è®­ç»ƒæ¨¡å‹å’Œåˆ†è¯å™¨...
09/23/2025 01:29:51 - INFO - root - âš¡ ä½¿ç”¨ Flash Attention
09/23/2025 01:29:51 - INFO - root - ğŸ§¬ è®­ç»ƒæŸ“è‰²ä½“: ['chr19']
09/23/2025 01:29:51 - INFO - root - ğŸ§¬ éªŒè¯æŸ“è‰²ä½“: ['chr19']
09/23/2025 01:29:51 - INFO - root - ğŸ·ï¸ è·å–æ•°æ®ç´¢å¼•...
09/23/2025 01:29:51 - INFO - root - ğŸ·ï¸ ç´¢å¼•æ–‡ä»¶å·²å­˜åœ¨ï¼Œç›´æ¥åŠ è½½: data/processed/all_chroms_train_index.csv
09/23/2025 01:29:51 - INFO - root - ğŸ·ï¸ ç´¢å¼•æ–‡ä»¶å·²å­˜åœ¨ï¼Œç›´æ¥åŠ è½½: data/processed/all_chroms_val_index.csv
09/23/2025 01:29:51 - INFO - root - ğŸ“ˆ è®­ç»ƒä¿¡å·è½¨é“ï¼ˆ1ä¸ªï¼‰: ['batch27']
09/23/2025 01:29:51 - INFO - root - ğŸ“ˆ éªŒè¯ä¿¡å·è½¨é“ï¼ˆ1ä¸ªï¼‰: ['batch27']
09/23/2025 01:29:51 - INFO - root - ğŸ§© åˆ›å»ºè®­ç»ƒæ•°æ®é›†...
09/23/2025 01:29:51 - INFO - root - ğŸ§© åˆ›å»ºéªŒè¯æ•°æ®é›†...
09/23/2025 01:29:51 - INFO - root - âœ… è®­ç»ƒ: 3,663 æ ·æœ¬, éªŒè¯: 37 æ ·æœ¬
09/23/2025 01:29:51 - INFO - root - ğŸ§  æ„å»ºä¸‹æ¸¸é¢„æµ‹å¤´...
09/23/2025 01:29:53 - INFO - root - gcc -pthread -B /opt/miniconda3/compiler_compat -fno-strict-overflow -DNDEBUG -O2 -Wall -fPIC -O2 -isystem /opt/miniconda3/include -fPIC -O2 -isystem /opt/miniconda3/include -fPIC -c /tmp/tmphymsgmt6/test.c -o /tmp/tmphymsgmt6/test.o
09/23/2025 01:29:53 - INFO - root - gcc -pthread -B /opt/miniconda3/compiler_compat /tmp/tmphymsgmt6/test.o -laio -o /tmp/tmphymsgmt6/a.out
09/23/2025 01:29:53 - INFO - root - gcc -pthread -B /opt/miniconda3/compiler_compat -fno-strict-overflow -DNDEBUG -O2 -Wall -fPIC -O2 -isystem /opt/miniconda3/include -fPIC -O2 -isystem /opt/miniconda3/include -fPIC -c /tmp/tmpknvpfro1/test.c -o /tmp/tmpknvpfro1/test.o
09/23/2025 01:29:53 - INFO - root - gcc -pthread -B /opt/miniconda3/compiler_compat /tmp/tmpknvpfro1/test.o -laio -o /tmp/tmpknvpfro1/a.out
09/23/2025 01:29:53 - INFO - root - ğŸ“Š æ¨¡å‹å‚æ•°é‡: 1,246,799,874 (å¯è®­ç»ƒ: 1,246,799,874 â†’ 100.0%)
09/23/2025 01:29:53 - INFO - root - âš™ï¸ é…ç½®è®­ç»ƒå‚æ•°...
09/23/2025 01:29:53 - INFO - root - gcc -pthread -B /opt/miniconda3/compiler_compat -fno-strict-overflow -DNDEBUG -O2 -Wall -fPIC -O2 -isystem /opt/miniconda3/include -fPIC -O2 -isystem /opt/miniconda3/include -fPIC -c /tmp/tmpsn0vqu54/test.c -o /tmp/tmpsn0vqu54/test.o
09/23/2025 01:29:53 - INFO - root - gcc -pthread -B /opt/miniconda3/compiler_compat -fno-strict-overflow -DNDEBUG -O2 -Wall -fPIC -O2 -isystem /opt/miniconda3/include -fPIC -O2 -isystem /opt/miniconda3/include -fPIC -c /tmp/tmpdsl483us/test.c -o /tmp/tmpdsl483us/test.o
09/23/2025 01:29:53 - INFO - root - gcc -pthread -B /opt/miniconda3/compiler_compat /tmp/tmpdsl483us/test.o -laio -o /tmp/tmpdsl483us/a.out
09/23/2025 01:29:53 - INFO - root - gcc -pthread -B /opt/miniconda3/compiler_compat /tmp/tmpsn0vqu54/test.o -laio -o /tmp/tmpsn0vqu54/a.out
09/23/2025 01:29:53 - INFO - root - gcc -pthread -B /opt/miniconda3/compiler_compat -fno-strict-overflow -DNDEBUG -O2 -Wall -fPIC -O2 -isystem /opt/miniconda3/include -fPIC -O2 -isystem /opt/miniconda3/include -fPIC -c /tmp/tmpafd605pa/test.c -o /tmp/tmpafd605pa/test.o
09/23/2025 01:29:53 - INFO - root - gcc -pthread -B /opt/miniconda3/compiler_compat -fno-strict-overflow -DNDEBUG -O2 -Wall -fPIC -O2 -isystem /opt/miniconda3/include -fPIC -O2 -isystem /opt/miniconda3/include -fPIC -c /tmp/tmpb36s7oer/test.c -o /tmp/tmpb36s7oer/test.o
09/23/2025 01:29:53 - INFO - root - gcc -pthread -B /opt/miniconda3/compiler_compat -fno-strict-overflow -DNDEBUG -O2 -Wall -fPIC -O2 -isystem /opt/miniconda3/include -fPIC -O2 -isystem /opt/miniconda3/include -fPIC -c /tmp/tmpzpoqhvoy/test.c -o /tmp/tmpzpoqhvoy/test.o
09/23/2025 01:29:53 - INFO - root - gcc -pthread -B /opt/miniconda3/compiler_compat /tmp/tmpafd605pa/test.o -laio -o /tmp/tmpafd605pa/a.out
09/23/2025 01:29:53 - INFO - root - gcc -pthread -B /opt/miniconda3/compiler_compat /tmp/tmpzpoqhvoy/test.o -laio -o /tmp/tmpzpoqhvoy/a.out
09/23/2025 01:29:53 - INFO - root - gcc -pthread -B /opt/miniconda3/compiler_compat /tmp/tmpb36s7oer/test.o -laio -o /tmp/tmpb36s7oer/a.out
09/23/2025 01:29:53 - INFO - root - gcc -pthread -B /opt/miniconda3/compiler_compat -fno-strict-overflow -DNDEBUG -O2 -Wall -fPIC -O2 -isystem /opt/miniconda3/include -fPIC -O2 -isystem /opt/miniconda3/include -fPIC -c /tmp/tmpd9vmcr2a/test.c -o /tmp/tmpd9vmcr2a/test.o
09/23/2025 01:29:53 - INFO - root - gcc -pthread -B /opt/miniconda3/compiler_compat /tmp/tmpd9vmcr2a/test.o -L/usr/local/cuda -L/usr/local/cuda/lib64 -lcufile -o /tmp/tmpd9vmcr2a/a.out
09/23/2025 01:29:53 - INFO - root - gcc -pthread -B /opt/miniconda3/compiler_compat -fno-strict-overflow -DNDEBUG -O2 -Wall -fPIC -O2 -isystem /opt/miniconda3/include -fPIC -O2 -isystem /opt/miniconda3/include -fPIC -c /tmp/tmppx24iks_/test.c -o /tmp/tmppx24iks_/test.o
09/23/2025 01:29:53 - INFO - root - gcc -pthread -B /opt/miniconda3/compiler_compat /tmp/tmppx24iks_/test.o -L/usr/local/cuda -L/usr/local/cuda/lib64 -lcufile -o /tmp/tmppx24iks_/a.out
09/23/2025 01:29:53 - INFO - root - gcc -pthread -B /opt/miniconda3/compiler_compat -fno-strict-overflow -DNDEBUG -O2 -Wall -fPIC -O2 -isystem /opt/miniconda3/include -fPIC -O2 -isystem /opt/miniconda3/include -fPIC -c /tmp/tmp5s7u_767/test.c -o /tmp/tmp5s7u_767/test.o
09/23/2025 01:29:53 - INFO - root - gcc -pthread -B /opt/miniconda3/compiler_compat /tmp/tmp5s7u_767/test.o -laio -o /tmp/tmp5s7u_767/a.out
09/23/2025 01:29:53 - INFO - root - gcc -pthread -B /opt/miniconda3/compiler_compat -fno-strict-overflow -DNDEBUG -O2 -Wall -fPIC -O2 -isystem /opt/miniconda3/include -fPIC -O2 -isystem /opt/miniconda3/include -fPIC -c /tmp/tmpimkf0kco/test.c -o /tmp/tmpimkf0kco/test.o
09/23/2025 01:29:53 - INFO - root - gcc -pthread -B /opt/miniconda3/compiler_compat -fno-strict-overflow -DNDEBUG -O2 -Wall -fPIC -O2 -isystem /opt/miniconda3/include -fPIC -O2 -isystem /opt/miniconda3/include -fPIC -c /tmp/tmpwgjranuw/test.c -o /tmp/tmpwgjranuw/test.o
09/23/2025 01:29:53 - INFO - root - gcc -pthread -B /opt/miniconda3/compiler_compat /tmp/tmpimkf0kco/test.o -L/usr/local/cuda -L/usr/local/cuda/lib64 -lcufile -o /tmp/tmpimkf0kco/a.out
09/23/2025 01:29:53 - INFO - root - gcc -pthread -B /opt/miniconda3/compiler_compat /tmp/tmpwgjranuw/test.o -L/usr/local/cuda -L/usr/local/cuda/lib64 -lcufile -o /tmp/tmpwgjranuw/a.out
09/23/2025 01:29:54 - INFO - root - gcc -pthread -B /opt/miniconda3/compiler_compat -fno-strict-overflow -DNDEBUG -O2 -Wall -fPIC -O2 -isystem /opt/miniconda3/include -fPIC -O2 -isystem /opt/miniconda3/include -fPIC -c /tmp/tmp3tg9pdxh/test.c -o /tmp/tmp3tg9pdxh/test.o
09/23/2025 01:29:54 - INFO - root - gcc -pthread -B /opt/miniconda3/compiler_compat /tmp/tmp3tg9pdxh/test.o -L/usr/local/cuda -L/usr/local/cuda/lib64 -lcufile -o /tmp/tmp3tg9pdxh/a.out
09/23/2025 01:29:54 - INFO - root - gcc -pthread -B /opt/miniconda3/compiler_compat -fno-strict-overflow -DNDEBUG -O2 -Wall -fPIC -O2 -isystem /opt/miniconda3/include -fPIC -O2 -isystem /opt/miniconda3/include -fPIC -c /tmp/tmpwab9_il_/test.c -o /tmp/tmpwab9_il_/test.o
09/23/2025 01:29:54 - INFO - root - gcc -pthread -B /opt/miniconda3/compiler_compat -fno-strict-overflow -DNDEBUG -O2 -Wall -fPIC -O2 -isystem /opt/miniconda3/include -fPIC -O2 -isystem /opt/miniconda3/include -fPIC -c /tmp/tmpnoe9e7u5/test.c -o /tmp/tmpnoe9e7u5/test.o
09/23/2025 01:29:54 - INFO - root - gcc -pthread -B /opt/miniconda3/compiler_compat /tmp/tmpwab9_il_/test.o -L/usr/local/cuda -L/usr/local/cuda/lib64 -lcufile -o /tmp/tmpwab9_il_/a.out
09/23/2025 01:29:54 - INFO - root - gcc -pthread -B /opt/miniconda3/compiler_compat /tmp/tmpnoe9e7u5/test.o -L/usr/local/cuda -L/usr/local/cuda/lib64 -lcufile -o /tmp/tmpnoe9e7u5/a.out
09/23/2025 01:29:54 - INFO - root - gcc -pthread -B /opt/miniconda3/compiler_compat -fno-strict-overflow -DNDEBUG -O2 -Wall -fPIC -O2 -isystem /opt/miniconda3/include -fPIC -O2 -isystem /opt/miniconda3/include -fPIC -c /tmp/tmp6juysz7q/test.c -o /tmp/tmp6juysz7q/test.o
09/23/2025 01:29:54 - INFO - root - gcc -pthread -B /opt/miniconda3/compiler_compat /tmp/tmp6juysz7q/test.o -L/usr/local/cuda -L/usr/local/cuda/lib64 -lcufile -o /tmp/tmp6juysz7q/a.out
09/23/2025 01:29:55 - INFO - root - ğŸ‹ï¸â€â™‚ï¸ å¯åŠ¨è®­ç»ƒ...
09/23/2025 03:03:45 - INFO - root - âœ… è®­ç»ƒå®Œæˆï¼
09/23/2025 03:03:45 - INFO - root - ğŸ§¹ èµ„æºå·²é‡Šæ”¾: train_datasetï¼ˆLazyGenomicDatasetï¼‰
09/23/2025 03:03:45 - INFO - root - ğŸ§¹ èµ„æºå·²é‡Šæ”¾: val_datasetï¼ˆLazyGenomicDatasetï¼‰
09/23/2025 03:03:45 - INFO - root - ğŸ§¹ wandb run å·²ç»“æŸ
09/23/2025 03:03:45 - INFO - root - ğŸ‰ ä¸»æµç¨‹æ‰§è¡Œå®Œæ¯•ï¼

```

## EvaluationÂ Index

### 2.1Â trajectoryÂ level/singleÂ baseÂ accuracyÂ index

[è¯·è‡³é’‰é’‰æ–‡æ¡£æŸ¥çœ‹ã€Œç”µå­è¡¨æ ¼ã€](https://alidocs.dingtalk.com/i/nodes/QG53mjyd80RbdBYktbZpMBm4V6zbX04v?iframeQuery=anchorId%3DX02mg3hpixkf97ys221l6&utm_medium=dingdoc_doc_plugin_card&utm_scene=person_space&utm_source=dingdoc_doc)

### 2.2Â GeneÂ LevelÂ Indicators

| pearson | log1pÂ pearson | spearman |
| --- | --- | --- |
| 0.8867 | 0.9553 | 0.6422 |

### 2.3Â differentialÂ geneÂ expression:Â predictedÂ versusÂ trueÂ trajectories

TheÂ numberÂ ofÂ genesÂ withÂ significantÂ differenceÂ (|Â log2(FoldChange)|>2)Â wasÂ 0.Â (AfterÂ zscoreÂ standardization)

### ![image.png](https://alidocs.oss-cn-zhangjiakou.aliyuncs.com/res/1wvqreb4KwLGenak/img/8188e6d0-2420-4c61-9cdf-b0f5722b22ee.png)![image.png](https://alidocs.oss-cn-zhangjiakou.aliyuncs.com/res/1wvqreb4KwLGenak/img/cd20eb99-3bee-4db7-8189-469a8da8d5b5.png)

### 2.4Â trajectoryÂ visualization

*   **AccurateÂ consistencyÂ ofÂ forecasts**
    

ChromosomeÂ 19Â global:Â TrueÂ inÂ blue,Â predictedÂ inÂ Red

![image.png](https://alidocs.oss-cn-zhangjiakou.aliyuncs.com/res/1wvqreb4KwLGenak/img/8df97a24-138c-419b-a283-e0edf879e9c5.png)

ZFP36ï¼š

![image.png](https://alidocs.oss-cn-zhangjiakou.aliyuncs.com/res/1wvqreb4KwLGenak/img/3ae87bb5-5d4e-405b-af01-2999b018e80d.png)

JUNBï¼š

![image.png](https://alidocs.oss-cn-zhangjiakou.aliyuncs.com/res/1wvqreb4KwLGenak/img/8cbc4d93-f453-459b-821d-1a292d19ada6.png)

chr19:Â 10587331-11635907Â ï¼ˆAlphaGenomeåœ¨HepG2Â CellÂ Lineä¸­é€‰å–çš„åŒºåŸŸï¼‰

![image.png](https://alidocs.oss-cn-zhangjiakou.aliyuncs.com/res/1wvqreb4KwLGenak/img/0acce3fd-9dc3-4087-bfd5-110673d740ab.png)

### CaseÂ 2:Â OomicsÂ +Â TextÂ InteractiveÂ DiseaseÂ Diagnosis

## ProjectÂ Overview

ThisÂ projectÂ isÂ aÂ DNAÂ sequenceÂ analysisÂ systemÂ basedÂ onÂ DNAÂ modelÂ +Â textÂ model,Â whichÂ isÂ mainlyÂ usedÂ forÂ geneÂ variationÂ effectÂ predictionÂ andÂ diseaseÂ associationÂ analysis.Â TheÂ projectÂ combinesÂ aÂ DNAÂ sequenceÂ encoderÂ withÂ aÂ large-scaleÂ languageÂ model,AchieveÂ aÂ deepÂ understandingÂ andÂ analysisÂ ofÂ DNAÂ sequences.

![image](https://alidocs.oss-cn-zhangjiakou.aliyuncs.com/res/meonarb8PVbQ0qXx/img/e26295ae-704f-47e4-a304-d81961de6dbf.jpg)

## InputÂ andÂ outputÂ analysis

### 1.1Â InputÂ dataÂ format

#### 1.1.1Â KEGGÂ dataÂ set

*   **FileÂ Format**:Â JSONÂ format
    
*   **DataÂ structure**:
    

```markdown
  ```json
  {
    "question": "æŸ“è‰²ä½“ä¿¡æ¯å’Œé€šè·¯ç½‘ç»œå®šä¹‰çš„é—®é¢˜æè¿°",
    "answer": "ç–¾ç—…åç§°ï¼ˆå¦‚ï¼šcushing syndrome, parkinson's disease, amyotrophic lateral sclerosisï¼‰",
    "reasoning": "è¯¦ç»†çš„æ¨ç†æ­¥éª¤ï¼ŒåŒ…å«10ä¸ªæ­¥éª¤çš„ç”Ÿç‰©å­¦åˆ†æ",
    "reference_sequence": "å‚è€ƒDNAåºåˆ—ï¼ˆå¤§å†™å­—æ¯ï¼Œå»é™¤ç©ºæ ¼ï¼‰",
    "variant_sequence": "å˜å¼‚DNAåºåˆ—ï¼ˆå¤§å†™å­—æ¯ï¼Œå»é™¤ç©ºæ ¼ï¼‰"
  }
  ```
```

#### 1.1.2Â VEP(VariantÂ EffectÂ Prediction)Â dataÂ set

*   **FileÂ Format**:Â JSONÂ format
    
*   **DataÂ structure**:
    

```markdown
  ```json
  {
    "question": "å˜å¼‚æ•ˆåº”é¢„æµ‹é—®é¢˜",
    "answer": "å˜å¼‚æ•ˆåº”åˆ†ç±»ç»“æœ",
    "reference_sequence": "å‚è€ƒåºåˆ—",
    "variant_sequence": "å˜å¼‚åºåˆ—"
  }
  ```
```

### 1.2Â outputÂ dataÂ format

#### 1.2.1Â ModelÂ Output

*   **TextÂ Generation**:Â TheÂ modelÂ generatesÂ textÂ thatÂ containsÂ theÂ reasoningÂ processÂ andÂ theÂ finalÂ answer.
    
*   **Format**:
    

```markdown
  ```
  <|im_start|>assistant
  [æ¨ç†å†…å®¹]
  Answer: [æœ€ç»ˆç­”æ¡ˆ]<|im_end|>
  ```
```

#### 1.2.2Â EvaluationÂ Results

**CSVÂ format**:Â containsÂ theÂ comparisonÂ betweenÂ theÂ predictionÂ resultÂ andÂ theÂ realÂ label.

*   **ColumnÂ structure**:
    
*   'Ground\_truth':Â RealÂ label
    
*   'Pred\_Label':Â predictionÂ label
    
*   'Generated\_text':Â CompleteÂ generatedÂ text
    

## DataÂ preprocessingÂ process

### 2.1Â DNAÂ sequenceÂ preprocessing

#### 2.1.1Â SequenceÂ Normalization

**CaseÂ conversion**:Â convertÂ allÂ DNAÂ sequencesÂ toÂ uppercaseÂ letters

*   **SpaceÂ removal**:Â RemovesÂ whiteÂ spaceÂ charactersÂ fromÂ theÂ sequence
    
*   **SequenceÂ truncation**:Â UseÂ theÂ 'truncate\_DNAÂ 'functionÂ toÂ truncateÂ theÂ sequenceÂ fromÂ bothÂ ends
    

```markdown
 ```python
  def truncate_dna(example, truncate_dna_per_side=1024):
      # ä»åºåˆ—ä¸¤ç«¯å„æˆªæ–­1024ä¸ªç¢±åŸºå¯¹
      # å¦‚æœåºåˆ—å¤ªçŸ­ï¼Œè¿”å›ä¸­é—´éƒ¨åˆ†
  ```
```

#### 2.1.2Â SequenceÂ Tokenization

*   **OURGENÂ \*\*\*\*Â taggers**:Â UseÂ character-levelÂ taggersÂ toÂ processÂ DNAÂ sequences
    
*   **SpecialÂ tag**:Â AddÂ '<|Â dna\_startÂ |>',Â '<|Â dna\_padÂ |>',Â '<|Â dna\_endÂ |>'Â tags
    
*   **SequenceÂ lengthÂ limit**:Â TheÂ maximumÂ lengthÂ isÂ 2048Â tokens.
    

### 2.2Â TextÂ preprocessing

#### 2.2.1Â ConversationÂ FormatÂ Conversion

*   **MultimodalÂ Format**:Â ConversationÂ formatÂ thatÂ combinesÂ DNAÂ sequencesÂ andÂ text
    
*   **RoleÂ Definitions**:
    
*   'user':Â containsÂ DNAÂ sequencesÂ andÂ questions
    
*   'assistant':Â containsÂ reasoningÂ processÂ andÂ answers
    

#### 2.2.2Â TemplateÂ Application

*   **ChatÂ Template**:Â UseÂ aÂ customÂ chatÂ templateÂ toÂ formatÂ input
    
*   **SpecialÂ tagÂ handling**:Â CorrectlyÂ handleÂ '<|Â im\_startÂ |>'Â andÂ '<|Â im\_endÂ |>'Â tags
    

### 2.3Â DataÂ loadingÂ andÂ batchÂ processing

#### 2.3.1Â DataÂ SetÂ Segmentation

*   **TrainingÂ set**:Â 80%
    
*   **ValidationÂ Set**:Â 10%
    
*   **TestÂ set**:Â 10%
    

#### 2.3.2Â BatchÂ Functions

*   **qwen\_DNAÂ \_collate\_FN**:Â aÂ batchÂ functionÂ designedÂ specificallyÂ forÂ QwenÂ DNAÂ Models
    
*   **LabelÂ mask**:Â CalculateÂ theÂ lossÂ onlyÂ forÂ theÂ helperÂ replyÂ part
    
*   **FillÂ processing**:Â leftÂ fillÂ policy,AddÂ SpecialÂ Marker
    

## EvaluationÂ indicatorÂ scheme

### 3.1Â classificationÂ indicators

#### 3.1.1Â BasicÂ indicators

*   **AccuracyÂ (Accuracy)**:Â theÂ proportionÂ ofÂ correctlyÂ predictedÂ samplesÂ toÂ theÂ totalÂ sample
    
*   **Precision**:Â MacroÂ averageÂ Precision,AverageÂ ofÂ accuracyÂ ratesÂ forÂ allÂ categories
    
*   **RecallÂ rateÂ (Recall)**:Â macroÂ averageÂ RecallÂ rate,Â theÂ averageÂ ofÂ allÂ categoriesÂ ofÂ RecallÂ rate
    

**F1Â scoreÂ (F1-Score)**:Â MacroÂ averageÂ F1Â score,Â reconciliationÂ averageÂ ofÂ precisionÂ andÂ recall

#### 3.1.2Â calculationÂ method

```python
# ä½¿ç”¨sklearnçš„classification_report
report_dict = classification_report(
    y_true, y_pred, 
    labels=labels, 
    output_dict=True, 
    zero_division=1
)

# æå–å®å¹³å‡æŒ‡æ ‡
macro_metrics = report_dict['macro avg']
Accuracy = accuracy_score(ground_truth, pred_label)
Precision = macro_metrics['precision']
Recall = macro_metrics['recall'] 
F1_score = macro_metrics['f1-score']

```

### 3.2Â AssessmentÂ Process

#### 3.2.1Â ValidationÂ PhaseÂ Assessment

*   **Real-timeÂ generation**:Â GenerateÂ textÂ onÂ theÂ validationÂ set
    

**AnswerÂ Extraction**:Â ExtractÂ TheÂ AnswerÂ partÂ fromÂ theÂ generatedÂ text

*   **IndicatorÂ calculation**:Â calculatesÂ categoryÂ indicatorsÂ andÂ recordsÂ themÂ inÂ theÂ log.
    

#### 3.2.2Â TestÂ PhaseÂ Assessment

*   **CompleteÂ evaluation**:Â CompleteÂ evaluationÂ onÂ theÂ testÂ set
    
*   **SaveÂ Results**:Â SaveÂ theÂ predictionÂ resultsÂ asÂ aÂ CSVÂ file.
    
*   **IndicatorÂ Summary**:Â GenerateÂ aÂ detailedÂ classificationÂ report
    

### 3.3Â SpecialÂ treatment

#### 3.3.1Â AnswerÂ extraction

*   **RegularÂ Matching**:Â UseÂ theÂ 'extract\_single\_entryÂ 'functionÂ toÂ extractÂ theÂ answer
    
*   **FormatÂ Processing**:Â ProcessingÂ theÂ predictionÂ resultÂ containingÂ theÂ '
    

| model\_type | Model | Source | Accuracy | F1-score | Precision | Recall |
| --- | --- | --- | --- | --- | --- | --- |
| dna | NT | Paper | 86.55% | 69.76% | 73.23% | 66.62% |
| dna | NT | Code | 89.31% | 68.99% | 74.20% | 68.14% |
| dna | our\_gene | Code | 91.72% | 78.75% | 81.09% | 81.64% |
| dna | evo2 | Code | 76.90% | 44.94% | 47.58% | 45.99% |
| dna | evo2 | Paper | 88.28% | 72.43% | 75.23% | 69.83% |
| dna-llm | NT+Qwen1B | Paper | 88.42% | 72.13% | 75.42% | 71.91% |
| dna-llm | NT+Qwen4B | Paper | 96.90% | 89.03% | 90.99% | 89.38% |
| dna-llm | NT+Qwen1B | Code | 93.45% | 90.26% | 97.88% | 88.47% |
| dna-llm | NT+Qwen4B | Code | 97.24% | 95.35% | 100.00% | 94.07% |
| dna-llm | our\_gene+Qwen1B | Code | 91.72% | 86.39% | 99.33% | 82.49% |
| dna-llm | our\_gene+Qwen4B | Code | 99.31% | 92.93% | 98.48% | 93.94% |
| dna-llm | evo2+Qwen1B | Code | 85.97% | 74.72% | 94.29% | 70.51% |
| dna-llm | evo2+Qwen4B | Code | 92.63% | 86.29% | 97.17% | 84.29% |
| dna-llm | evo2+Qwen3-1B | Paper | 90.42% | 75.62% | 77.42% | 73.91% |
| dna-llm | evo2+Qwen3-4B | Paper | 97.24% | 86.30% | 86.75% | 87.25% |
| llm | Qwen1B | Paper | 85.17% | 65.71% | 71.39% | 64.19% |
| llm | Qwen4B | Paper | 93.48% | 85.44% | 88.31% | 86.72% |
| llm | Qwen1B | Code | 88.62% | 82.56% | 93.03% | 80.77% |
| llm | Qwen4B | Code | 97.93% | 93.06% | 98.99% | 93.41% |

## 8.Â LicenceÂ description

ThisÂ workÂ advocatesÂ theÂ openÂ sharingÂ ofÂ biologicalÂ AIÂ communityÂ andÂ supportsÂ MITÂ License.Â SeeÂ 'MITÂ©2025Â '

## 9.Â ContactÂ Us

IfÂ youÂ haveÂ questionsÂ orÂ cooperationÂ intention,FeelÂ freeÂ toÂ contactÂ us.Â Email:Â bgi-Genos@genomics.cn
